packages/libs/client/src/signing/chainweaver/signWithChainweaver.ts

```diff
@@ -14,75 +14,101 @@ import { parseTransactionCommand } from '../utils/parseTransactionCommand';
-  const response = await fetch('http://127.0.0.1:9467/v1/quicksign', {
-    method: 'POST',
-    body,
-    headers: { 'Content-Type': 'application/json;charset=utf-8' },
-  });
+    const response = await fetch(`${chainweaverUrl}/v1/quicksign`, {
+      method: 'POST',
+      body,
+      headers: { 'Content-Type': 'application/json;charset=utf-8' },
+    });
 
-  const bodyText = await response.text();
+    const bodyText = await response.text();
 
-  // response is not JSON when not-ok, that's why we use try-catch
-  try {
-    const result = JSON.parse(bodyText) as IQuicksignResponse;
+    // response is not JSON when not-ok, that's why we use try-catch
+    try {
+      const result = JSON.parse(bodyText) as IQuicksignResponse;
 
-    if ('error' in result) {
-      if ('msg' in result.error) {
-        console.log('error in result', result.error.msg);
+      if ('error' in result) {
+        if ('msg' in result.error) {
+          console.log('error in result', result.error.msg);
+        }
+        throw new Error(JSON.stringify(result.error));
       }
-      throw new Error(JSON.stringify(result.error));
-    }
 
-    result.responses.map((signedCommand, i) => {
-      transactions[i] = addSignatures(
-        transactions[i],
-        ...signedCommand.commandSigData.sigs.filter(isASigner),
+      result.responses.map((signedCommand, i) => {
+        transactions[i] = addSignatures(
+          transactions[i],
+          ...signedCommand.commandSigData.sigs.filter(isASigner),
+        );
+      });
+
+      return isList ? transactions : transactions[0];
+    } catch (error) {
+      throw new Error(
+        'An error occurred when adding signatures to the command' +
+          `\nResponse from v1/quicksign was \`${bodyText}\`. ` +
+          `\nCode: \`${response.status}\`` +
+          `\nText: \`${response.statusText}\` ` +
+          `${error}`,
       );
-    });
+    }
+  }) as ISignFunction;
 
-    return isList ? transactions : transactions[0];
-  } catch (error) {
-    throw new Error(
-      'An error occurred when adding signatures to the command' +
-        `\nResponse from v1/quicksign was \`${bodyText}\`. ` +
-        `\nCode: \`${response.status}\`` +
-        `\nText: \`${response.statusText}\` ` +
-        `${error}`,
-    );
-  }
-}) as ISignFunction;
+/**
+ * * Lets you sign with chainweaver according to {@link https://github.com/kadena-io/KIPs/blob/master/kip-0015.md | sign-v1 API}
+ *
+ * @deprecated Use {@link createSignWithChainweaver} instead
+ * @public
+ */
+export const signWithChainweaver: ISignFunction = signTransactions(
+  'http://127.0.0.1:9467',
+);
+
+/**
+ * Creates the signWithChainweaver function with interface {@link ISignFunction}
+ * Lets you sign with chainweaver according to {@link https://github.com/kadena-io/KIPs/blob/master/kip-0015.md | sign-v1 API}
+ *
+ * @public
+ */
+export function createSignWithChainweaver(
+  options = { chainweaverUrl: 'http://127.0.0.1:9467' },
+): ISignFunction {
```

Comment:
Minor comment, maybe use just "host" and add an example to the @params

```suggestion
/**
 * Creates the signWithChainweaver function with interface {@link ISignFunction}
 * Lets you sign with Chainweaver according to {@link https://github.com/kadena-io/KIPs/blob/master/kip-0015.md | sign-v1 API}
 *
 * @param options - object to customize behaviour.
 *   - `host: string` - the host of the chainweaver instance to use. Defaults to `http://127.0.0.1:9467`
 * @returns - {@link ISignFunction}
 */
export function createSignWithChainweaver(
  options = { host: 'http://127.0.0.1:9467' },
): ISignFunction {
```